pkg/container-disk/container-disk.go

```diff
@@ -398,7 +399,29 @@ func ExtractImageIDsFromSourcePod(vmi *v1.VirtualMachineInstance, sourcePod *kub
 		imageIDs[KernelBootVolumeName] = vmi.Spec.Domain.Firmware.KernelBoot.Container.Image
 	}
 
-	for _, status := range sourcePod.Status.ContainerStatuses {
+	containersToCheck := append([]kubev1.ContainerStatus{}, sourcePod.Status.ContainerStatuses...)
+	if imageVolumeEnabled {
+		// When imageVolume is Enabled, there are two cases:
+		// 1. First migration: the Pod was created before the digest was known, so the volumes do not include it.
+		//    We include the initContainerStatuses in containersToCheck so that the loop below can extract the digest
+		//    from the statuses and update imageIDs (similar to how we handle standard containerDisk sidecars When imageVolume is disabled).
```

Comment:
Maybe I'm missing something, but I don't understand why in the original flow (without image volumes) we don't need to inspect init containers and in this case we do. Aren't no-op init containers added in the legacy container-disk case as well?